app/models/solidus_stripe/payment_intent.rb

```diff
@@ -5,41 +5,69 @@ class PaymentIntent < ApplicationRecord
     belongs_to :order, class_name: 'Spree::Order'
     belongs_to :payment_method, class_name: 'SolidusStripe::PaymentMethod'
 
-    def self.retrieve_stripe_intent(payment_method:, order:)
-      find_by(payment_method: payment_method, order: order)&.stripe_intent
-    end
+    def self.prepare_for_payment(payment)
+      # Find or create the intent for the payment.
+      intent =
+        find_by(payment_method: payment.payment_method, order: payment.order) ||
+          new(payment_method: payment.payment_method, order: payment.order)
+            .tap { _1.update!(stripe_intent_id: _1.create_stripe_intent.id) }
 
-    def self.create_stripe_intent(payment_method:, order:, stripe_intent_options: {})
-      instance = new(payment_method: payment_method, order: order)
-      instance.create_stripe_intent(stripe_intent_options).tap { instance.update!(stripe_intent_id: _1.id) }
-    end
+      # Update the intent with the previously acquired payment method.
+      intent.payment_method.gateway.request {
+        Stripe::PaymentIntent.update(intent.stripe_intent_id, payment_method: payment.source.stripe_payment_method_id)
+      }
 
-    def stripe_intent
-      payment_method.gateway.request do
-        Stripe::PaymentIntent.retrieve(stripe_intent_id)
-      end
+      # Attach the payment intent to the payment.
+      payment.update!(response_code: intent.stripe_intent.id)
+
+      intent
     end
 
-    def create_payment!(amount: order.total, add_to_wallet: false)
-      payment = order.payments.create!(
-        state: 'pending',
+    def process_payment
+      payment = order.payments.valid.find_by!(
         payment_method: payment_method,
-        amount: amount,
         response_code: stripe_intent.id,
-        source: payment_method.payment_source_class.new(
-          payment_method: payment_method,
-        ),
       )
 
-      if add_to_wallet && order.user && stripe_intent.setup_future_usage.present?
-        payment.source.update!(stripe_payment_method_id: stripe_intent.payment_method)
-        order.user.wallet.add payment.source
+      payment.started_processing!
+
+      case stripe_intent.status
+      when 'requires_capture'
+        payment.pend! unless payment.pending?
+        successful = true
+      else
+        payment.failure!
+        successful = false
       end
 
-      payment
+      SolidusStripe::LogEntries.payment_log(
+        payment,
+        success: successful,
+        message: I18n.t("solidus_stripe.intent_status.#{stripe_intent.status}"),
+        data: stripe_intent,
+      )
+
+      if successful
+        order.complete!
+      else
+        order.payment_failed!
+      end
+
+      successful
+    end
+
+    def stripe_intent
+      @stripe_intent ||= payment_method.gateway.request do
+        Stripe::PaymentIntent.retrieve(stripe_intent_id)
+      end
+    end
+
+    def reload
```

Comment:
No one is using this method in this commit

Reply:
This was added because we thought it could be useful in the future in case we want to refresh a Payment Intent from Stripe.

Maybe we can move it to a separate commit?

Reply:
I see this case as similar to adding `method_missing` and not implementing `respond_to?`, I don't know if there's a better way or rails support to reset instance variables on reload that's definitely what we need if we use ivars inside an AR model.

Not much unlike what is done by AR itself (e.g. [here](https://github.com/rails/rails/blob/7c70791470fc517deb7c640bead9f1b47efb5539/activerecord/lib/active_record/autosave_association.rb#L233-L238)).

@kennyadsl I'm sort of ok with removing it but I'd like to hear your thoughts with the context provided above first 🙌

Reply:
Ah, I didn't think it's needed when we call the already existing `reload` method on the instance. If we need it to make the `reload` method work correctly for `PaymentIntent`, I definitely favor keeping it. Maybe it just belongs to a separate PR?

Reply:
But if it's needed for something we added here, it's totally ok to have it. If that's the case, I think it's just not super clear, so maybe a couple of lines in the commit description would be good, or a separate commit.